libraries/db/include/graphene/db/undo_database.hpp

```diff
@@ -59,17 +59,7 @@ namespace graphene { namespace db {
                {
                   mv._apply_undo = false;
                }
-               ~session() {
-                  try {
-                     if( _apply_undo ) _db.undo();
-                  }
-                  catch ( const fc::exception& e )
-                  {
-                     elog( "${e}", ("e",e.to_detail_string() ) );
-                     throw; // maybe crash..
-                  }
-                  if( _disable_on_exit ) _db.disable();
-               }
+               ~session(); // defined in implementation file to prevent repeated compiler warnings
```

Comment:
Can remove the comment, no excuse needed for not implementing methods in .hpp file

Reply:
Comment removed. Please also take a look at #1511 for another piece I would like to add to this PR.